src/test/java/com/google/sps/LoginServletTest.java

```diff
@@ -57,28 +72,60 @@ public void setUp() throws Exception {
       // Set up a fake HTTP response 
       responseWriter = new StringWriter();
       when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+      // Initialize datastore
+      datastore = DatastoreServiceFactory.getDatastoreService();
   }
 
   @After
   public void tearDown() throws Exception {
       helper.setEnvIsLoggedIn(true);
       helper.tearDown();
   }
+  
+  
+  @Test
+  public void userInDatastoreAlready() throws Exception {
+
+      datastore.put(USER_A.createEntity());
```

Comment:
Include a comment that explains what each test is checking. That way we can come back later and add edge cases without having to figure out what each test is doing.

Reply:
I think it's better to put that information into the name of the test function -- that way, when a test fails, the test runner will print the name of the function that failed.

In this case though, the function name could add additional information to help the reader understand the test. I would propose renaming the test to "loginWhenUserInDatastoreAlready".

(If the name of the function would be too long -- then write a comment.)

Reply:
Fixed in #5